### PR TITLE
cleanup(virtctl): Remove dryRun options

### PR DESCRIPTION
**Pull request by opokornyy**
**What this PR does / why we need it**:

This PR removes dryRun options and adds functionality to CleanNamespaces to remove VirtualMachinePreference and VirtualMachineClusterPreference instead. It also updates help message for this command.

**Release note**:
```
None
```

### CODE REVIEW

**Comment by 0xFelix** [comment]
Looks good to me, is this needed for Instancetypes too?

**Comment by opokornyy** [comment]
> Looks good to me, is this needed for Instancetypes too?

It is already done for Instancetypes.

File: tests/virtctl/create_preference.go [hunk 1]
```diff
@@ -36,13 +38,16 @@ var _ = Describe("[sig-compute] create preference", decorators.SigCompute, func(
 		case *instancetypev1alpha2.VirtualMachinePreference:
 			ExpectWithOffset(1, namespaced).To(BeTrue(), "expected VirtualMachinePreference to be created")
 			ExpectWithOffset(1, obj.Kind).To(Equal("VirtualMachinePreference"))
-			preference, err := virtClient.VirtualMachinePreference(util.NamespaceTestDefault).Create(context.Background(), (*instancetypev1alpha2.VirtualMachinePreference)(obj), metav1.CreateOptions{DryRun: []string{metav1.DryRunAll}})
+			preference, err := virtClient.VirtualMachinePreference(util.NamespaceTestDefault).Create(context.Background(), (*instancetypev1alpha2.VirtualMachinePreference)(obj), metav1.CreateOptions{})
```
Comment (lyarwood) [inline tests/virtctl/create_preference.go:41]: nit - I don't think you need to cast obj here.

File: tests/virtctl/create_preference.go [hunk 2]
```diff
@@ -36,13 +38,16 @@ var _ = Describe("[sig-compute] create preference", decorators.SigCompute, func(
...
 			ExpectWithOffset(1, err).ToNot(HaveOccurred())
 			return &preference.Spec, nil
 		case *instancetypev1alpha2.VirtualMachineClusterPreference:
 			ExpectWithOffset(1, namespaced).To(BeFalse(), "expected VirtualMachineClusterPreference to be created")
 			ExpectWithOffset(1, obj.Kind).To(Equal("VirtualMachineClusterPreference"))
-			clusterPreference, err := virtClient.VirtualMachineClusterPreference().Create(context.Background(), (*instancetypev1alpha2.VirtualMachineClusterPreference)(obj), metav1.CreateOptions{DryRun: []string{metav1.DryRunAll}})
+			obj.Labels = map[string]string{
```
Comment (lyarwood) [inline tests/virtctl/create_preference.go:47]: FWIW I bet this isn't happening for cluster instance types and preferences in other functional tests, something to look into outside of this PR.

**Review by lyarwood — COMMENTED** [review]
/lgtm

**Review by 0xFelix — APPROVED** [review]
/lgtm

**Review by xpivarc — COMMENTED** [review]
Looks good to me, lets just close the gaps we have here.

File: tests/virtctl/create_preference.go (hunk 2 above)
Comment (opokornyy) [reply tests/virtctl/create_preference.go:47]: I have added labels to these functions as well.

**Pull request merged**